stem/boundary.py

```diff
@@ -0,0 +1,96 @@
+from typing import List, Dict, Any, Union, Optional
+from dataclasses import dataclass, field
+from abc import ABC, abstractmethod
+
+
+@dataclass
+class BoundaryParametersABC(ABC):
+    """
+    Abstract base class for boundary parameters
+    """
+    pass
+
+    @abstractmethod
+    def is_constraint(self) -> bool:
+        """
+        Helping function to determine whether the boundary should be in the list of
+        constraint (True) or in the list of loads (False).
+        Returns:
+            bool
+        """
+        pass
+
+
+@dataclass
+class DisplacementConstraint(BoundaryParametersABC):
+    """
+    Class containing the boundary parameters for displacement constraint
+
+    Attributes:
+        active (List[bool]): Activate/deactivate constraint for each direction.
+        is_fixed (List[bool]): Specify if constraint is fixed.
+        value (List[float]):
+    """
+
+    active: List[bool] = field(default_factory=lambda: [True, True, True])
+    is_fixed: List[bool] = field(default_factory=lambda: [True, True, True])
+    value: List[float] = field(default_factory=lambda: [0.0, 0.0, 0.0])
+
+    def is_constraint(self) -> bool:
+        return True
+
+
+@dataclass
+class RotationConstraint(BoundaryParametersABC):
+    """
+    Class containing the boundary parameters for rotation constraint
+
+    Attributes:
+        active (List[bool]): Activate/deactivate constraint for each direction.
+        is_fixed (List[bool]): Specify if constraint is fixed.
+        value (List[float]):
+    """
+
+    active: List[bool] = field(default_factory=lambda: [True, True, True])
+    is_fixed: List[bool] = field(default_factory=lambda: [True, True, True])
+    value: List[float] = field(default_factory=lambda: [0.0, 0.0, 0.0])
+
+    def is_constraint(self) -> bool:
+        return True
+
+@dataclass
+class AbsorbingBoundary(BoundaryParametersABC):
+    """
+    Class containing the boundary parameters for a point boundary
+
+    Attributes:
+        absorbing_factors (List[float]): Activate/deactivate boundary for each direction.
```

Comment:
abosrbing factors are not no activate the boundary, These indicate how much of the P-wave and S-wave should be damped

Reply:
0, is no damping; 1: full damping; 0<x,1: partial damping

Reply:
Thanks for the explanation, will include it in the docstring!